lib/qesapdeployment.pm

```diff
@@ -745,4 +749,112 @@ sub qesap_get_az_resource_group {
     return $result;
 }
 
+=head3 qesap_calculate_az_address_range
+
+Calculate the vnet and subnet address
+ranges. The format is 10.ip2.ip3.0/21 and
+ /24 respectively. ip2 and ip3 are calculated
+ using the slot number as seed.
+
+=over 1
+
+=item B<SLOT> - integer to be used as seed in calculating addresses
+
+=back
+
+=cut
+
+sub qesap_calculate_az_address_range {
+    my %args = @_;
+    croak 'Missing mandatory slot argument' unless $args{slot};
+
+    die "Invalid 'slot' argument - valid values are 1-8192" if ($args{slot} > 8192 || $args{slot} < 1);
+
+    my $offset = ($args{slot} - 1) * 8;
+
+    # addresses are of the form 10.ip2.ip3.0/21 and /24 respectively
+    #ip2 gets incremented when is >=256
+    my $ip2 = int($offset / 256);
+    #ip3 gets incremented by 8 until it's >=256, then it resets
+    my $ip3 = $offset % 256;
+
+    my $vnet_address_range = sprintf("10.%d.%d.0/21", $ip2, $ip3);
+    my $subnet_address_range = sprintf("10.%d.%d.0/24", $ip2, $ip3);
+
+    return (
+        vnet_address_range => $vnet_address_range,
+        subnet_address_range => $subnet_address_range,
+    );
```

Comment:
Suggestion:

1. Remove L781-L783.
2. Replace L784-L787 with:
```
    return (
        vnet_address_range => sprintf("10.%d.%d.0/21", $ip2, $ip3),
        subnet_address_range => sprintf("10.%d.%d.0/24", $ip2, $ip3),
    )
```
No need for the extra variables IMO.